tests/unit/test_action.py

```diff
@@ -0,0 +1,72 @@
+"""Test Action abstraction for v0x01 and v0x04."""
+import unittest
+
+from pyof.foundation.basic_types import UBInt16, UBInt32
+
+from napps.kytos.of_core.v0x01.flow import Action as Action01
+from napps.kytos.of_core.v0x04.flow import Action as Action04
+
+
+class TestAction(unittest.TestCase):
+    """Tests for the Action class."""
+
+    def test_all_actions(self):
+        """Test all action fields from and to dict."""
+        action_dicts = [
+            {
+                'tests': [Action01, Action04],
+                'action': {
+                    'action_type': 'output',
+                    'port': UBInt32(1),
+                },
+            },
+            {
+                'tests': [Action01, Action04],
+                'action': {
+                    'action_type': 'set_vlan',
+                    'vlan_id': UBInt16(2),
+                },
+            },
+            {
+                'tests': [Action04],
+                'action': {
+                    'action_type': 'set_queue',
+                    'queue_id': UBInt32(4),
+                }
+            },
+            {
+                'tests': [Action04],
+                'action': {
+                    'action_type': 'pop_vlan',
+                }
+            },
+            {
+                'tests': [Action04],
+                'action': {
+                    'action_type': 'push_vlan',
+                    'tag_type': 'c',
+                }
+            },
+            {
+                'tests': [Action04],
+                'action': {
+                    'action_type': 'push_vlan',
+                    'tag_type': 's',
+                }
+            },
+        ]
+        for action_class in Action01, Action04:
+            with self.subTest(action_class=action_class):
```

Comment:
@ajoaoff I've realized that with `pytest` and `tox` with `subTest` even if when there are multiple failures, it only shows the first one, however if you run with `unittest.main` (without pytest) then it shows correctly if there are multiple failing subtests. Do you know how we can enhance the current execution to possibly show multiple failures with subtest?  Example:

- with `unittest`:


```

======================================================================
ERROR: test_all_actions (__main__.TestAction) (action_class=<class 'napps.kytos.of_core.v0x04.flow.Action'>, expected={'action_type': 'pop_vlanx'})
Test all action fields from and to dict.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/unit/test_action.py", line 65, in test_all_actions
    action = action_class.from_dict(expected)
  File "/home/viniarck/repos/napps/kytos/of_core/flow.py", line 308, in from_dict
    action_class = cls._action_class[action_type]
KeyError: 'pop_vlanx'

======================================================================
ERROR: test_all_actions (__main__.TestAction) (action_class=<class 'napps.kytos.of_core.v0x04.flow.Action'>, expected={'action_type': 'push_vlanx', 'tag_type': 's'})
Test all action fields from and to dict.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/unit/test_action.py", line 65, in test_all_actions
    action = action_class.from_dict(expected)
  File "/home/viniarck/repos/napps/kytos/of_core/flow.py", line 308, in from_dict
    action_class = cls._action_class[action_type]
KeyError: 'push_vlanx'

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=2)
```

- with `pytest`:

```

======================================================================================== FAILURES ========================================================================================
______________________________________________________________________________ TestAction.test_all_actions _______________________________________________________________________________

self = <of_core.tests.unit.test_action.TestAction testMethod=test_all_actions>

    def test_all_actions(self):
        """Test all action fields from and to dict."""
        action_dicts = [
            {
                'tests': [Action01, Action04],
                'action': {
                    'action_type': 'output',
                    'port': UBInt32(1),
                },
            },
            {
                'tests': [Action01, Action04],
                'action': {
                    'action_type': 'set_vlan',
                    'vlan_id': UBInt16(2),
                },
            },
            {
                'tests': [Action04],
                'action': {
                    'action_type': 'set_queue',
                    'queue_id': UBInt32(4),
                }
            },
            {
                'tests': [Action04],
                'action': {
                    'action_type': 'pop_vlanx',
                }
            },
            {
                'tests': [Action04],
                'action': {
                    'action_type': 'push_vlan',
                    'tag_type': 'c',
                }
            },
            {
                'tests': [Action04],
                'action': {
                    'action_type': 'push_vlanx',
                    'tag_type': 's',
                }
            },
        ]
        for action_class in Action01, Action04:
            with self.subTest(action_class=action_class):
                for action_test in action_dicts:
                    if action_class not in action_test['tests']:
                        continue
                    expected = action_test['action']
                    with self.subTest(expected=expected):
>                       action = action_class.from_dict(expected)

tests/unit/test_action.py:65:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'napps.kytos.of_core.v0x04.flow.Action'>, action_dict = {'action_type': 'pop_vlanx'}

    @classmethod
    def from_dict(cls, action_dict):
        """Build the proper Action from a dictionary.

        Args:
            action_dict (dict): Action attributes.
        """
        action_type = action_dict.get('action_type')
>       action_class = cls._action_class[action_type]
E       KeyError: 'pop_vlanx'

../napps/kytos/of_core/flow.py:308: KeyError
==================================================================================== warnings summary ====================================================================================
.direnv/python-3.6.15/src/kytos/kytos/core/config.py:173
  /home/viniarck/repos/of_core/.direnv/python-3.6.15/src/kytos/kytos/core/config.py:173: UserWarning: Unknown arguments: ['tests/', '-k', 'test_all_actions']
    warnings.warn(f"Unknown arguments: {unknown}")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================ short test summary info =================================================================================
FAILED tests/unit/test_action.py::TestAction::test_all_actions - KeyError: 'pop_vlanx'
```

Reply:
I'll merge this PR as it is and then try to move from `subTest` to  `pytest.parametrize` in another PR.